File: backend/app/schemas/queue.py
import datetime

from pydantic import BaseModel


class QueueRecipient(BaseModel):
    address: str
    delay_reason: str | None = None


class QueueEntry(BaseModel):
    queue_id: str
    queue_name: str
    arrival_time: datetime.datetime
    message_size: int
    sender: str
    recipients: list[QueueRecipient]

File: frontend/src/lib/api/queue.ts
import { apiFetch } from "../apiClient";

export interface QueueRecipient {
  address: string;
  delay_reason: string | null;
}

export interface QueueEntry {
  queue_id: string;
  queue_name: string;
  arrival_time: string;
  message_size: number;
  sender: string;
  recipients: QueueRecipient[];
}

export function listQueue(): Promise<QueueEntry[]> {
  return apiFetch<QueueEntry[]>("/api/queue");
}

export function retryQueueMessage(queueId: string): Promise<void> {
  return apiFetch<void>(`/api/queue/${encodeURIComponent(queueId)}/retry`, { method: "POST" });
}

export function deleteQueueMessage(queueId: string): Promise<void> {
  return apiFetch<void>(`/api/queue/${encodeURIComponent(queueId)}`, { method: "DELETE" });
}
